### PR TITLE
Add doc about deploying the DHCP daemonset with Multus

### PR DESCRIPTION
**Pull request by thomasferrandiz**
Also remove doc about customizing the whereabouts image as it can lead to users deploying an image version not compatible with our own chart.

Related issue: https://github.com/rancher/rke2/issues/3917

### CODE REVIEW

**Review by rbrtbnfgl — APPROVED** [review]

File: docs/install/network_options.md [hunk 1]
```diff
@@ -218,16 +218,13 @@ spec:
...
+To use any of these plugins, a proper NetworkAttachmentDefinition object will need to be created to define the configuration of the secondary network. The definition is then referenced by pod annotations, which Multus will use to provide extra interfaces to that pod. An example using the macvlan cni plugin with Multus is available [in the multus-cni repo](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/quickstart.md#storing-a-configuration-as-a-custom-resource).
 
-## Using Multus with the Whereabouts CNI
-
-[Whereabouts](https://github.com/k8snetworkplumbingwg/whereabouts) is an IP Address Management (IPAM) CNI plugin that assigns IP addresses cluster-wide.
-Starting with RKE2 1.22, RKE2 includes the option to use Whereabouts with Multus to manage the IP addresses of the additional interfaces created through Multus.
-In order to do this, you need to use [HelmChartConfig](../helm.md#customizing-packaged-components-with-helmchartconfig) to configure the Multus CNI to use Whereabouts.
+## Using Multus with the DHCP daemonset
```
Comment (brandond) [inline docs/install/network_options.md:223]: Will this be backported to the older branches? If so, we usually indicate that by mentioning the release month and all the releases in which it is first available:
https://github.com/rancher/rke2-docs/blob/478c18c0a424aa65e864d842acb905bd17615762/docs/install/requirements.md?plain=1#L19

Comment (thomasferrandiz) [reply docs/install/network_options.md:223]: yes it has been backported.

**Review by brandond — APPROVED** [review]

**Comment by manuelbuil** [comment]
I'm concerned that the docs become too long and unorganized, which might confuse users, especially if they are not familiar with how Multus works. I think it would be beneficial if we created a new subsection for multus specifying the three IPAM plugins we support:
* host-local
* whereabouts
* dhcp-daemon

For example adding a section: "Multus IPAM options" and then three tabs, as we have with Canal, Cilium and Calico here: https://docs.rke2.io/install/network_options#install-a-cni-plugin.

For host-local, for example.:
```
host-local IPAM plugin allocates ip addresses out of a set of address ranges. It stores the state locally on the host filesystem, therefore ensuring uniqueness of IP addresses on a single host. Therefore, we don't recommend it for multi-node clusters. This IPAM plugin does not require any extra deployment. For more information: https://www.cni.dev/plugins/current/ipam/host-local/ 
```
For whereabours, we can use what we already have, which explains what it is and how to deploy it
For dhcp-daemon, we can use what you are providing in this PR

What do you think? I think it helps because then people understand that whereabouts, dhcp-daemon and host-local are alternatives for the same purpose and they don't get overwhelmed by too many multus subsections

**Comment by thomasferrandiz** [comment]
@manuelbuil good point, I can do that it will be clearer

**Review by manuelbuil — APPROVED** [review]

**Pull request merged**